app/assets/javascripts/actions/tickets_actions.js

```diff
@@ -120,13 +121,40 @@ export const readAllMessages = ticket => async (dispatch) => {
   dispatch({ type: SET_MESSAGES_TO_READ, data: json });
 };
 
+export const dateSegments = (
+  tomorrow = moment().add(1, 'day'),
+  length = 5, // Number of dates to retrieve
+  segement = 15 // Number of days between requests
+) => {
+  return Array.from({ length }).map((_el, index) => {
+    const days = index * segement;
+    return moment(tomorrow).subtract(days, 'days').format('YYYY-MM-DD');
+  });
+};
+
+const fetchSomeTickets = async (dispatch, before, after) => {
+  let query = `created_before=${before}`;
+  if (after) query += `&created_after=${after}`;
+
+  const response = await fetch(`/td/tickets?${query}`);
+  return response.json().then(({ tickets }) => {
+    dispatch({ type: RECEIVE_TICKETS, data: tickets });
+  });
+};
+
+// Fetch as many tickets as possible
 export const fetchTickets = () => async (dispatch) => {
   dispatch({ type: FETCH_TICKETS });
 
-  const response = await fetch('/td/tickets');
-  const json = await response.json();
+  const segments = dateSegments();
```

Comment:
Locally, I have 107 tickets, I end up with more than 400 in `state.tickets.all`. Also seeing some duplicate key warnings in some cases, so it looks like each ticket is ending up represented multiple times in the set.

Reply:
Oh, of course, that's because I haven't updated TicketDispenser.